code/game/gamemodes/umbrage/umbrage_datum.dm

```diff
@@ -0,0 +1,136 @@
+//Handles everything related to psi and lucidity.
+/datum/umbrage
+	var/datum/mind/linked_mind
+	var/mob/living/linked_body
+	var/psi = 100 //Our psi, used for abilities.
+	var/max_psi = 100
+	var/psi_regeneration = 20 //The limit for psi regenerated during the cycle.
+	var/psi_used_since_last_cycle //How much psi we've used in the last five seconds.
+	var/cycle_progress = 0 //When this reaches 5, it will reset to 0 and regenerate up to 20 spent psi.
+	var/lucidity = 3 //Lucidiy is used to buy abilities. We gain one each time we drain someone.
+	var/lucidity_drained = 3 //How much lucidity we've taken overall.
+	var/list/linked_abilities = list() //Abilities that we have
+	var/list/linked_ability_types = list() //Used for cloning and gaining new bodies
+	var/list/drained_minds = list() //Minds eaten with Devour Will.
```

Comment:
yes.
if the list isn't there when you try to add it, it's built, this is the principle of lazyloading, you build data structures (something typically memory hungry because... it holds data.. in memory...) as and when you need them, and kill them off when you don't anymore (unless they have a costly creation/destruction)

Think of this example with how buckling used to be, every single atom had a buckled mobs list, pre initialised, that's 65,000 odd atoms, so 65,000 odd lists, how many of those are chais? maybe 100-200 at best, how many are vehicles? maybe 20 at most, how many are bucklable mobs? maybe 5

so we're using 225/65,000 lists, at best, what are the other 64,775 lists doing? hogging memory for no reason!!